Libplanet.Store/Trie/MerkleTrie.Diff.cs

```diff
@@ -0,0 +1,160 @@
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using Bencodex.Types;
+using Libplanet.Store.Trie.Nodes;
+
+namespace Libplanet.Store.Trie
+{
+    public partial class MerkleTrie : ITrie
+    {
+        /// <inheritdoc cref="ITrie.Diff"/>
+        public IEnumerable<(KeyBytes Path, IValue? TargetValue, IValue SourceValue)>
+            Diff(ITrie other)
+        {
+            if (Root is null)
+            {
+                yield break;
+            }
+
+            var queue = new Queue<(INode, Nibbles)>();
+            queue.Enqueue((Root, new Nibbles(ImmutableArray<byte>.Empty)));
+
+            while (queue.Count > 0)
+            {
+                (INode node, Nibbles path) = queue.Dequeue();
+                INode? target = other.GetNode(path);
```

Comment:
If there is a performance issue, I guess this part will be such a point.

Reply:
I do have some vague idea on how to optimize this using a lexicographically ordered trie traversal method. 🙄

Reply:
Maybe I think it will be a similar idea. I left this comment for note 😀 I'll approve.